modules/product/src/main/java/com/opengamma/strata/product/swap/type/FixedInflationSwapTemplate.java

```diff
@@ -0,0 +1,439 @@
+/**
+ * Copyright (C) 2016 - present by OpenGamma Inc. and the OpenGamma group of companies
+ *
+ * Please see distribution for license.
+ */
+package com.opengamma.strata.product.swap.type;
+
+import java.io.Serializable;
+import java.time.LocalDate;
+import java.time.Period;
+import java.util.Map;
+import java.util.NoSuchElementException;
+import java.util.Set;
+
+import org.joda.beans.Bean;
+import org.joda.beans.BeanDefinition;
+import org.joda.beans.ImmutableBean;
+import org.joda.beans.JodaBeanUtils;
+import org.joda.beans.MetaProperty;
+import org.joda.beans.Property;
+import org.joda.beans.PropertyDefinition;
+import org.joda.beans.impl.direct.DirectFieldsBeanBuilder;
+import org.joda.beans.impl.direct.DirectMetaBean;
+import org.joda.beans.impl.direct.DirectMetaProperty;
+import org.joda.beans.impl.direct.DirectMetaPropertyMap;
+
+import com.opengamma.strata.basics.BuySell;
+import com.opengamma.strata.basics.date.Tenor;
+import com.opengamma.strata.basics.market.ReferenceData;
+import com.opengamma.strata.basics.market.ReferenceDataNotFoundException;
+import com.opengamma.strata.product.TradeTemplate;
+import com.opengamma.strata.product.swap.SwapTrade;
+
+/**
+ * A template for creating Inflation swap trades.
+ * <p>
+ * This defines almost all the data necessary to create a Inflation single currency {@link SwapTrade}.
```

Comment:
Should be "an" instead of "a". Also I'm not sure "Inflation" needs to be capitalised.